docs/source/add_new_model.rst

```diff
@@ -0,0 +1,811 @@
+   retrieve intermediate values. Is the original repository written in Tensorflow 1? Then you might have to rely on
+   TensorFlow print operations like https://www.tensorflow.org/api_docs/python/tf/print to output intermediate values.
+   Is the original repository written in Jax? Then make sure that the model is **not jitted** when running the forward
+   pass, *e.g.* check-out `this link <https://github.com/google/jax/issues/196>`_.
+-  Use the smallest pretrained checkpoint you can find. The smaller the checkpoint, the faster your debug cycle
+   becomes. It is not efficient if your pretrained model is so big that your forward pass takes more than 10 seconds.
+   In case only very large checkpoints are available, it might make more sense to create a dummy model in the new
+   environment with randomly initialized weights and save those weights for comparison with the 🤗 Transformers version
+   of your model
+-  Make sure you are using the easiest way of calling a forward pass in the original repository. Ideally, you want to
+   find the function in the original repository that **only** calls a single forward pass, *i.e.* that is often called
+   `predict`, `evaluate`, `forward` or `__call__`. You don't want to debug a function that calls `forward` multiple
+   times, *e.g.* to generate text, like `autoregressive_sample`, `generate`.
+-  Try to separate the tokenization from the model's `forward` pass. If the original repository shows examples where
+   you have to input a string, then try to find out where in the forward call the string input is changed to input ids
+   and start from this point. This might mean that you have to possibly write a small script yourself or change the
+   original code so that you can directly input the ids instead of an input string.
+-  Make sure that the model in your debugging setup is **not** in training mode, which often causes the model to yield
+   random outputs due to multiple dropout layers in the model. Make sure that the forward pass in your debugging
+   environment is **deterministic** so that the dropout layers are not used. Or use `transformers.file_utils.set_seed`
+   if the old and new implementations are in the same framework.
+
+The following section gives you more specific details/tips on how you can do this for *brand_new_bert*.
+
+Implement BrandNewBert into 🤗 Transformers
+-----------------------------------------------------------------------------------------------------------------------
+
+Next, you can finally add code to 🤗 Transformers. Go into the clone of your 🤗 Transformers’ fork:
+
+::
+
+   cd transformers
+
+In the special case that you are adding a model whose architecture exactly matches the model architecture of an
+existing model you only have to add a conversion script as described in `this section <#write-a-conversion-script>`__.
+In this case, you can just re-use the whole model architecture of the already existing model.
+
```

Comment:
FWIW, while porting I did a lot of:
```
cp modeling_bart.py modeling_fsmt.py
perl -pi -e 's|Bart|FSMT|g' modeling_fsmt.py
```
So basically building on the existing code and just renaming the classes.

Reply:
Yes! This should be done automatically now with the cookie-cutter tool